mantidproject/mantid
# Improve HB3A's IDF for calibration

Description of work.

In this PR, a mistake on interpreting the rows and columns in XML file written by SPICE is corrected. The IDF of HB3A is modified such that the distance between the sample and detector plane can be set via log file.  The detector pixel that the neutron beam hits is moved from the geometry center of the detector to pixel (128, 115). 

**To test:**
1. Check the unit tests please.  All the new features are tested in the unit tests.
2. Load any HB3A data, for example, http://neutron.ornl.gov/user_data/hb3a/exp455/Datafiles/HB3A_exp455_scan0085_0018.xml.  If you view the detector from source in the instrument view, then you should see the data same to the image from that web page.
3. You may try to set an arbitrary value to sample-detector offset value to check how it is applied.

Fixes #14087.

[Release notes](/docs/source/release/)



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
